src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+            system.assertEquals(contact.MailingStreet, addr.MailingStreet__c);
+            system.assertEquals(contact.MailingCity, addr.MailingCity__c);
+            system.assertEquals(true, addr.Default_Address__c);
+        }
+    }
+        
+    /*********************************************************************************************************
+    @description Update existing default Addresses.  
+    verify: Contact Mailing address fields updated
+    **********************************************************************************************************/            
+    @isTest 
+    public static void updateDefaultAddr() {
+        configSettings();
+
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // now let's update the Addresses
+        for (Integer i = 0; i < contactsAddrs.addrs.size(); i++) {
+            Address__c addr = contactsAddrs.addrs[i];
+            addr.MailingStreet__c = 'New Street' + i;
+            addr.MailingCity__c = 'New City' + i;
+        }
+        system.assertEquals(false, ADDR_Addresses_TDTM.hasRunAddrTrigger);
+        Test.startTest();
+        update contactsAddrs.addrs;
+        Test.stopTest();
+    
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c
+        from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.contains('New Street'));
+            system.assert(contact.MailingCity.contains('New City'));
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+    }
+    
+    /*********************************************************************************************************
+    @description Delete existing default Addresses.  
+    verify: Contact Mailing address fields cleared.
+    **********************************************************************************************************/            
+    @isTest 
+    static void deleteDefaultAddr() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+       
+        // now let's delete the Addresses
+        system.assertEquals(false, ADDR_Addresses_TDTM.hasRunAddrTrigger);
+        Test.startTest();
+        delete contactsAddrs.addrs;
+        Test.stopTest();
+    
+        // verify that the Contact address fields are cleared
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.